### PR TITLE
feat: donwload all button

### PR DESCRIPTION
**Pull request by tomwisecodes**
Partial implementation of
https://figma.com/design/60B9J4YPWyzytpFNhyoQgx/Downloads?node-id=70-7129&node-type=canvas&t=BHMZ8hdO8TOm4HeW-0

### CODE REVIEW

File: apps/nextjs/src/app/api/aila-download-all/route.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  await kv.set(taskId, "loading");
+
+  const zipStream = new PassThrough();
+  const archive = archiver("zip", { zlib: { level: 9 } });
+  archive.pipe(zipStream);
+
+  let filesProcessed = 0;
+
+  for (const { fileId, formats } of fileIdsAndFormats) {
+    if (!fileId || !Array.isArray(formats)) continue;
```
Comment (mantagen) [inline apps/nextjs/src/app/api/aila-download-all/route.ts:109]: Shouold we error here?

File: apps/nextjs/src/app/api/aila-download-all/route.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+        await saveDownloadEvent({
+          lessonExportId: lessonExport.id,
+          downloadedBy: userId,
+          ext,
+        });
+
+        filesProcessed++;
+      } catch (error) {
+        Sentry.captureException(error, { level: "error" });
```
Comment (mantagen) [inline apps/nextjs/src/app/api/aila-download-all/route.ts:146]: What is the effect here? the file does not end up in the zip?

Comment (mantagen) [reply apps/nextjs/src/app/api/aila-download-all/route.ts:146]: If so maybe 
```ts
const err = new Error("Error downloading file, not included in zip");
err.cause = error;
Sentry.ca...
```

File: apps/nextjs/src/app/api/aila-download-all/route.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+  if (!userId) {
+    const error = new Error("Download attempt without userId");
+    Sentry.captureException(error, { level: "warning" });
+    return new Response("Unauthorized", { status: 401 });
+  }
+
+  if (!fileIdsParam) {
+    return new Response("Invalid or missing fileIds", { status: 400 });
+  }
```
Comment (mantagen) [inline apps/nextjs/src/app/api/aila-download-all/route.ts:89]: Shall we sentry breadcrumb with lessonTitle and fileIdParam

File: apps/nextjs/src/app/api/aila-download-all/route.ts [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+      Sentry.captureException(error, { level: "warning" });
+      continue;
+    }
+
+    for (const ext of formats) {
+      try {
+        const res = await downloadDriveFile({ fileId, ext });
+
+        if ("error" in res) {
+          Sentry.captureException(res.error);
```
Comment (mantagen) [inline apps/nextjs/src/app/api/aila-download-all/route.ts:126]: Same question here as below

File: apps/nextjs/src/components/AppComponents/download/DownloadAllButton.tsx [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+                key === "lessonSlides" ? ["pptx", "pdf"] : ["docx", "pdf"],
+            };
+          }
+        })
+      : undefined;
+
+    const taskId = `download-all-${JSON.stringify(fileIdsAndFormats)}`;
+
+    function trackDownload(resourceFileType: ResourceFileTypeValueType) {
+      track.lessonPlanResourcesDownloaded({
```
Comment (mantagen) [inline apps/nextjs/src/components/AppComponents/download/DownloadAllButton.tsx:102]: Maybe double check with @emmalanches about this?

Comment (tomwisecodes) [reply apps/nextjs/src/components/AppComponents/download/DownloadAllButton.tsx:102]: Have checked in and we are adding a new download type

File: apps/nextjs/src/components/ExportsDialogs/useExportAllLessonAssets.ts [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+      const res = exportSlidesFullLessonSchema.safeParse(lesson);
+      setParseResult(res);
+    }
+  }, [lesson, setParseResult, active]);
+
+  const start = useCallback(() => {
+    if (!active) {
+      return;
+    }
+    console.log("STARTING");
```
Comment (mantagen) [inline apps/nextjs/src/components/ExportsDialogs/useExportAllLessonAssets.ts:48]: can lose the logs

File: apps/nextjs/src/components/ExportsDialogs/useExportAllLessonAssets.ts [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+  const query = trpc.exports.generateAllAssetExports.useMutation();
+
+  const [parseResult, setParseResult] = useState<
+    | { data?: LessonSlidesInputData; success: boolean; error?: ZodError }
+    | undefined
+  >({ success: false });
+  const debouncedParseResult = useDebounce(parseResult, 500);
+
+  useEffect(() => {
+    console.log("Use Export Lesson Slides");
```
Comment (mantagen) [inline apps/nextjs/src/components/ExportsDialogs/useExportAllLessonAssets.ts:37]: can lose

**Review by mantagen — APPROVED** [review]
Code looks good -- couple logs left in there

**Pull request merged**